core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,204 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for CollectionLinearizerService.
+ */
+
+describe('Collection linearizer service', function() {
+  var CollectionObjectFactory = null;
+  var CollectionNodeObjectFactory = null;
+  var CollectionLinearizerService = null;
+
+  var firstCollectionNode = null;
+  var secondCollectionNode = null;
+  var thirdCollectionNode = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    CollectionObjectFactory = $injector.get('CollectionObjectFactory');
+    CollectionNodeObjectFactory = $injector.get('CollectionNodeObjectFactory');
+    CollectionLinearizerService = $injector.get('CollectionLinearizerService');
+
+    var firstCollectionNodeBackendObject = {
+      exploration_id: 'exp_id0',
+      prerequisite_skills: [],
+      acquired_skills: ['exp title0'],
```

Comment:
I don't think any of these nodes should have prerequisite_skills or acquired_skills pre-populated. Your linearizer tests would be much stronger if you verified it properly sets up these prerequisite and acquired skills, since that's its core functionality. You'll need to initialize these in ``_createLinearCollection`` but that's only used for tests requiring an initial state. Most should be able to operate without prerequisite and acquired skills.